src/utils/events.ts

```diff
@@ -1,164 +1,153 @@
-): Promise<void> =>
-  await eventListenerGenerator(
-    eventSignature,
-    contractAddress,
-    ClientType.NetworkClient,
-  );
-
-/*
- * Colony Client specific event listener,
- * which uses `eventListenerGenerator` under the hood
- */
-export const addColonyEventListener = async (
-  eventSignature: ContractEventsSignatures,
-  contractAddress: string,
-): Promise<void> =>
-  await eventListenerGenerator(
-    eventSignature,
-    contractAddress,
-    ClientType.ColonyClient,
-  );
-
-/*
- * Token Client specific event listener,
- * which uses `eventListenerGenerator` under the hood
- */
-export const addTokenEventListener = async (
-  eventSignature: ContractEventsSignatures,
-  contractAddress: string,
-): Promise<void> =>
-  await eventListenerGenerator(
-    eventSignature,
-    contractAddress,
-    ClientType.TokenClient,
-  );
+  return filter;
+};
 
 /**
- * Extension specific event listener
- * It creates a new interface with ABI data describing
- * possible event signature
+ * Add the contract event to the event queue to be processed by the event processor.
  */
-export const addExtensionEventListener = async (
-  eventSignature: ContractEventsSignatures,
-  extensionAddress: string,
-): Promise<void> => {
-  const { provider } = networkClient;
-  const extensionContract = new Contract(extensionAddress, [
-    `event ${eventSignature}`,
-  ]);
-  const filter = {
-    topics: [utils.id(eventSignature)],
-    address: extensionAddress,
-  };
-
-  verbose(
-    'Added listener for Event:',
-    eventSignature,
-    extensionAddress ? `filtering Address: ${extensionAddress}` : '',
-  );
-
-  provider.on(filter, async (log: Log) => {
-    const event = await mapLogToContractEvent(log, extensionContract.interface);
+export const addEventToQueue = async (
+  client: NetworkClients,
+  clientType: ClientType,
+  contractAddress: string,
+  log: Log,
+  provider: Provider,
+) => {
+  const {
+    transactionHash,
+    logIndex,
+    blockNumber,
+    address: eventContractAddress,
+  } = log;
+  try {
+    const { hash: blockHash, timestamp } = await provider.getBlock(blockNumber);
 
-    if (event) {
-      addEvent(event);
+    let event = {
+      blockNumber,
+      transactionHash,
+      logIndex,
+      contractAddress,
+      blockHash,
+      timestamp,
+    };
+
+    switch (clientType) {
+      case ClientType.TokenClient: {
+        client = await getTokenClient(eventContractAddress, provider);
+        event.contractAddress = eventContractAddress;
+        addEvent({
+          ...client.interface.parseLog(log),
+          ...event,
+        });
+        break;
+      }
+
+      case ClientType.VotingReputationClient: {
+        /*
+         * In the case of Voting Rep, only add the event to the queue if the colony address
+         * associated with the event is the same as the colony address the listener is associated with.
+         */
+        const { colonyId } = await query('getColonyExtension', {
```

Comment:
Using this to get the colony the voting reputation extension is installed in. This is a guard against the motion being written to the db with the wrong colony (which is possible when the behaviour I'm seeing in development occurs, where the event fires on all colonies)